rattan-core/src/cells/mod.rs

```diff
@@ -362,3 +501,91 @@ macro_rules! wait_until_started {
 pub use crate::core::CALIBRATED_START_INSTANT as TRACE_START_INSTANT;
 #[cfg(feature = "first-packet")]
 pub use crate::core::FIRST_PACKET_INSTANT as TRACE_START_INSTANT;
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum State {
+    // Drops all packets
+    Drop,
+    // Passes through all packets
+    PassThrough,
+    // Normal operation
+    Normal,
+}
+
+pub struct AtomicState(AtomicU8);
```

Comment:
Why is this only used in the delay cell, not in other cells? This is not consistent.

Also, since you defined `State`, I think we should modify the `change_state` function signature to use `State`? Also, rename `State` to `CellState` to make it more understandable